### PR TITLE
Add IIFO ForageSharing

### PR DESCRIPTION
**Pull request by E-Mac27**
# Summary
Marked as Draft because it depends on #128.

Server will now ping each client at the start of an IIFO turn if they want to share their most recent foraging attempts. This information will be sent out to whoever they stated. 

Clients will be given the optional choice to implement their client functions (MakeForageInfo() and ReceiveForageInfo(...)). 
- MakeForageInfo() if this is unimplemented, the ShareForageInformation struct will be filled with nil values, meaning nothing will be sent. 
- ReceiveForageInfo(...) does not affect the Clients whether or not its implemented.

## Test Plan

Tested in internal/server/iifo_test.go
Also tested manually by filling out internal/team client.go

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

**Comment by mpardalos** [comment]
@E-Mac27 you can mark this as ready for review, #128 was merged.

File: internal/common/shared/forage.go [hunk 1]
```diff
@@ -18,3 +18,21 @@ type ForageDecision struct {
...
+	SharedFrom ClientID
+}
+
+// MakeForagingDict is a map of clients' -> foraging decisions, their resource obtained, and which
+// clients to send this information to
+type MakeForagingDict = map[ClientID]ShareForageInfomation
+
+// ReceiveForagingDict is a map of clients -> array of information of other clients
+// foraging decisions, their resource obtained, and which clients to sent this information
+type ReceiveForagingDict = map[ClientID][]ShareForageInfomation
```
Comment (lhl2617) [inline internal/common/shared/forage.go]: Can we name this with `Foragxxx`?

Comment (lhl2617) [reply internal/common/shared/forage.go]: Actually, should we change the naming for foraging stuff to follow this one? I think that's a stretch, but at least the types here are `Foragxxx`

Comment (lhl2617) [reply internal/common/shared/forage.go]: > Actually, should we change the naming for _all_ foraging stuff to follow this one? I think that's a stretch, but at least the types here are `Foragxxx`

Comment (E-Mac27) [reply internal/common/shared/forage.go]: Not entirely too sure what to change MakeForagingDict or ReceiveForagingDict to such that its still readable.

Comment (lhl2617) [reply internal/common/shared/forage.go]: `ForagingOfferDict` and `ForagingReceiptDict`?

Comment (E-Mac27) [reply internal/common/shared/forage.go]: Done

**Review by NGovani — APPROVED** [review]
Looks good

**Review by NGovani — APPROVED** [review]

**Pull request merged**